### DISCUSSION TITLE
Create as many environment as servers defined in `OpenAPI` file during import

### DISCUSSION THREAD

**Discussion by sixdouglas** [Kong/insomnia#3627] [Ideas]
**Is your feature request related to a problem? Please describe.**
I'm always frustrated when I import OpenAPI descriptor file containing multiple servers and only the first one is considered as Insomnia Environment.

**Describe the solution you'd like**
I wish there would be as many environment created in the Insomnia workspace as servers defined in the file.

**Describe alternatives you've considered**

**Additional context**
I may be able to help on this feature. Please let me know if this feature would be accepted.

**Comment by MartinThoma** [comment]
This would be really nice!

**Comment by develohpanda** [comment]
>  I may be able to help on this feature. Please let me know if this feature would be accepted.

Hi @sixdouglas, I just saw this comment, my apologies! This is certainly something we'd like to solve and PRs would be welcome. There's some additional information in https://github.com/Kong/insomnia/issues/4234 as well.

**Comment by sixdouglas** [comment]
@develohpanda here is a PR https://github.com/Kong/insomnia/pull/4254 for this issue.

Any remarks is welcomed.

**Comment by ryanholden8** [comment]
Not sure where to request this but it would also be great if `server.variables` were also imported as environment variables. - https://swagger.io/specification/#server-object

**Reply by develohpanda** [reply]
This is definitely related but a new Discussion would be the way to go! There might be a GitHub Issue or Discussion open for it already

**Comment by sixdouglas** [comment]
@develohpanda here is a new version of my implementation.

**Comment by ajsaclayan** [comment]
Hi all,

I'm seeing a similar issue where I'm defining a different server for a specific route (https://swagger.io/docs/specification/api-host-and-base-path/ under Overriding Servers), but Kong points the route to the global list. Should I open a new discussion?

**Reply by wdawson** [reply]
Thanks @ajsaclayan , I think a separate discussion would be best

**Reply by ThaDaVos** [reply]
Where's this seperate discussion at? As I am also experiencing this (as I have a microservices environment and written a small Koa based Node service to combine the OpenAPI's from all services into one for ease of use and importing)

**Reply by ThaDaVos** [reply]
Just noticed, Postman does do this - set the correct URL - but I prefer insomnia over postman
